Lib/test/test_xmlrpc.py

```diff
@@ -1171,6 +1171,70 @@ def test_gzip_decode_limit(self):
         xmlrpclib.gzip_decode(encoded, max_decode=-1)
 
 
+class HeadersServerTestCase(BaseServerTestCase):
+    class RequestHandler(xmlrpc.server.SimpleXMLRPCRequestHandler):
+        test_headers = None
+
+        def do_POST(self):
+            self.__class__.test_headers = self.headers
+            return super().do_POST()
+    requestHandler = RequestHandler
+    standard_headers = [
+        'Host', 'Accept-Encoding', 'Content-Type', 'User-Agent',
+        'Content-Length']
+
+    def setUp(self):
+        self.RequestHandler.test_headers = None
+        return super().setUp()
+
+    def test_header(self):
+        p = xmlrpclib.ServerProxy(URL, headers=[('X-Test', 'foo')])
+        self.assertEqual(p.pow(6, 8), 6**8)
+
+        headers = self.RequestHandler.test_headers
+        self.assertListEqual(
+            sorted(headers.keys()), sorted(self.standard_headers + ['X-Test']))
+        self.assertEqual(headers.get('X-Test'), 'foo')
+
+    def test_header_many(self):
+        p = xmlrpclib.ServerProxy(
+            URL, headers=[('X-Test', 'foo'), ('X-Test-Second', 'bar')])
+        self.assertEqual(p.pow(6, 8), 6**8)
+
+        headers = self.RequestHandler.test_headers
+        self.assertListEqual(
+            sorted(headers.keys()),
+            sorted(self.standard_headers + ['X-Test', 'X-Test-Second']))
```

Comment:
I think maybe the best thing to do is to factor this out into a separate assertion function:

```python
def assertContainsAdditionalHeaders(self, headers, additional):
    expected_keys = sorted(self.standard_headers + list(additional.keys())
    self.assertListEqual(sorted(headers.keys()), expected_keys)

    for key in additional.keys():
        self.assertEqual(headers.get(key), additional[key])
```

Then you can use it everywhere like this:

```python
self.asssertContainsAdditionalHeaders(headers, {'X-Test': 'foo'})
```

Reply:
Indeed it is cleaner.